src/interactive/workspace.ts

```diff
@@ -1,45 +1,77 @@
 import * as vscode from 'vscode';
+import * as repl from './repl';
 
 let g_terminal: vscode.Terminal = null
-let g_replVariables: string = '';
 
-export class REPLTreeDataProvider implements vscode.TreeDataProvider<string> {
-    private _onDidChangeTreeData: vscode.EventEmitter<string | undefined> = new vscode.EventEmitter<string | undefined>();
-    readonly onDidChangeTreeData: vscode.Event<string | undefined> = this._onDidChangeTreeData.event;
+interface WorkspaceVariable {
+    name: string,
+    type: string,
+    value: string
+}
+
+let g_replVariables: WorkspaceVariable[] = undefined;
```

Comment:
I think this just can be `let g_replVariables: WorkspaceVariable[] = []` and remove the condition check at L27, reuse `clearVariables` at L74-L75 etc